File: backend/app/v1/modules/block/dto/block_dto.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.v1.modules.block.dto.block_type_dto import BlockTypeDTO


class BlockDTO(BaseModel):
    """
    Represents a learning content block.

    A block belongs to either:
    - a Topic (topic-level introduction)
    - a Section (section-level content)
    """

    model_config = ConfigDict(
        from_attributes=True
    )

    id: UUID

    type: BlockTypeDTO

    content: str

    display_order: int

    topic_id: UUID | None = None

    section_id: UUID | None = None

    created_at: datetime

    updated_at: datetime

File: frontend/quantum-mind-ui/src/app/features/learn/interfaces/block.dto.ts
export type BlockType =
  | 'paragraph'
  | 'heading'
  | 'image'
  | 'code'
  | 'equation';


export interface BlockDTO {
    id: string;

    type: BlockType;

    content: string;

    display_order: number;

    topic_id: string | null;

    section_id: string | null;

    created_at: string;

    updated_at: string;
}
